src/Core/File/LightData.hpp

```diff
@@ -43,54 +56,121 @@ class RA_CORE_API LightData : public AssetData {
             linear( l ),
             quadratic( q ) {}
     };
-
-    /// CONSTRUCTOR
-    LightData( const std::string& name = "", const LightType& type = UNKNOWN );
-
-    LightData( const LightData& data ) = default;
-
-    /// DESTRUCTOR
+    // TODO : allow to define other attenuation function such \max\left(0,1-\frac{\left|x\right|^2}{r^2}\right)^2
+
+    /**
+     * Light constructor.
+     * Must be calle explicitely with a name for the light and its type.
+     */
+    explicit LightData( const std::string& name = "", const LightType& type = UNKNOWN );
+
+    /**
+     * Copy constructor.
+     * As a Light a a union like object, no default copy constructor could be generated. This will take care of copying the good members of the union.
+     * @param data
+     */
+    LightData( const LightData& data );
+
+    /// Destructor
     ~LightData();
 
-    /// NAME
+    /**
+     * Extension of of the AssetData interface so that we can modify an abcet afeter ts creation.
+     * This has no impact on the coherence of the LightData object and could be used without restriction.
+     */
     inline void setName( const std::string& name );
 
-    /// TYPE
-    inline LightType getType() const;
-    inline void setType( const LightType& type );
 
-    /// FRAME
+    /**
+     * Acces to the local frame of the light.
+     * @return the local frame
+     */
     inline Core::Matrix4 getFrame() const;
```

Comment:
return reference here ?

Reply:
no need for reference nor const reference. Copy elision and rvalue optimisation is now well supported by all the C++ compiler. So, there will be always at most one copy with both the ref or object return version of such a function. And, in this case, it is clear that the returned Matrix is a rvalue.

Reply:
Ok, it's just that I don't like to rely on copy elision, but it is a matter of style I guess.
In debug mode the matrix will be copied anyway.

Reply:
Why ??? rvalues are handled independently of the compilation mode no ? (well, not sure with visualC++ compiler but ...)
And we can find lot of pertinent reasons to return an object and not a ref. More, in this case, the ref must be constant as the method is marked as const. and so the copy will be done.
for recommandations and explanation (by Stroustrup and Sutter - and even the whole C++ comunity) see https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f15-prefer-simple-and-conventional-ways-of-passing-information.

Reply:
According to https://en.cppreference.com/w/cpp/language/copy_elision#Notes :
>  Because some compilers do not perform copy elision in every situation where it is allowed (e.g., in debug mode), programs that rely on the side-effects of copy/move constructors and destructors are not portable.

Reply:
So for me it should be 
```c++
inline const Core::Matrix4& getFrame() const;
```

Reply:
Well, It is not because some compilers do not generate portable programs (depending on the debug/or release mode) that we must write some over-engineered code. 
If we have to program according to the way a (unknown) compiler might understand our code instead of the way the langage is constructed to make the programmer concentrate on its concepts rather than on the code generation (that, if i recall well, is the job of the compiler), we are not "sorti de l'auberge :p  )
That being said, **i give up. You won** (even if i disagree). 
Definitively I personally prefer B. Stroustrup and H. Sutter over VisualC++ :p